pkg/storage/stores/indexshipper/downloads/table.go

```diff
@@ -0,0 +1,347 @@
+package downloads
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/dskit/concurrency"
+	"github.com/pkg/errors"
+	"go.etcd.io/bbolt"
+
+	"github.com/grafana/loki/pkg/storage/chunk"
+	chunk_util "github.com/grafana/loki/pkg/storage/chunk/util"
+	"github.com/grafana/loki/pkg/storage/stores/indexshipper/index"
+	"github.com/grafana/loki/pkg/storage/stores/shipper/storage"
+	util_log "github.com/grafana/loki/pkg/util/log"
+)
+
+// timeout for downloading initial files for a table to avoid leaking resources by allowing it to take all the time.
+const (
+	downloadTimeout        = 5 * time.Minute
+	maxDownloadConcurrency = 50
+)
```

Comment:
should these be configurable?

Reply:
I try to avoid adding new configurations since we have more than 1000 of them now. This value has been working fine across all our cells ranging from tiny to too big. The dynamics with the `tsdb` index might change, and we can add it later?

Reply:
`I try to avoid adding new configurations since we have more than 1000 of them now.` Fair, can we at least include in docs somewhere (if there's a spot where we describe how the shipper works) that these are the values used?